Added totals option to report building

bcornwellmott opened the pull request:
This is a very rough attempt to add a totals row to the report builder, which seems to be a common request on the forum. I'm really not sure if this is a "correct" way to do it, but it seems to work. A few notes: 
1. It only applies to the columns that are visible (e.g if only the first 20 rows are shown, it will sum only those 20 rows)
2. You access it by clicking the "Toggle Totals Row" button in the report builder.
3. It sums anything it can interpret as a number (including some status')

As it's very rough, I'd encourage improvements on it. However, it's a tool that is badly needed in ERP.

RobertSchouten committed:
tag catagories

RobertSchouten committed:
filter_dashboard

RobertSchouten committed:
updating sidebar stats to list_sidebar

bcornwellmott committed:
Added totals option to report building

bcornwellmott committed:
Added totals option to report building

rmehta committed:
[fix] tests

rmehta committed:
[fix] call update_db after customize form if changed

rmehta committed:
[fix] integration scheduler events

RobertSchouten committed:
assorted cleanup and v7.1 fixes

rmehta committed:
[fix] integration scheduler events

nabinhait committed:
get the default value of check field if none, for insert / update

rmehta committed:
[fix] integration scheduler events

rmehta committed:
[fix] [hot] fixes grid frappe/erpnext#6646

rmehta commented:
@bcornwellmott thanks, will look into this next week

netchampfaris committed:
[fix] hotkey fix for Safari (#2175)

rmehta commented on frappe/public/js/frappe/views/reports/reportview.js:
```diff
@@ -501,6 +504,17 @@ frappe.views.ReportView = frappe.ui.Listing.extend({
+	make_totals_row_button: function() {
+		var me = this;
+
+		this.column_picker = new frappe.ui.ColumnPicker(this);
```
is this here by mistake :)

bcornwellmott replied:
Yes, I think you're right.

rmehta commented:
Will try and test this. Maybe we can also add it as a property that can be saved with the report!

bcornwellmott committed:
Removed accidentally included column_picker

nabinhait committed:
bumped to version 7.0.46

nabinhait committed:
Merge branch 'hotfix'

nabinhait committed:
Fixed version conflict

rmehta committed:
[fix] sortable now needs group

vjFaLk committed:
Added new commands and cleanup (#2183)

Updated Cheatsheet

vjFaLk committed:
[minor] fixed links in portugese translations

saurabh6790 committed:
[minor][fix] remove raise import error (#2186)

KanchanChauhan committed:
Disables reports will not show up in list view

netchampfaris committed:
[gantt] show popover on hover, code cleanup (#2177)

* [gantt] show popover on hover, code cleanup

* [gantt] use variables instead of hex

rohitwaghchaure committed:
[Documentation] Fixed images in how to make report

KanchanChauhan committed:
Zambia currency updated

nabinhait committed:
Merge pull request #2187 from KanchanChauhan/hide-disabled-reports

Disables reports will not show up in list view

nabinhait committed:
Merge pull request #2190 from KanchanChauhan/zambia-currency-update

Zambia currency updated

nabinhait committed:
Merge pull request #2188 from rohitwaghchaure/query_report_documentation

[Documentation] Fixed images in how to make report

rmehta committed:
[fix] link selection inside grid, fixes bug caused by Stortable

netchampfaris committed:
[gantt] minor (#2189)

revant committed:
OAuth 2 Provider for frappe (#2167)

* OAuth 2.0 Classes Added

Added oauthlib in requirement.txt
Added class WebApplicationServer for Authorization code grant and Bearer tokens.
Added class OAuthWebRequestValidator for Validating Oauth Request for Web Application

* copied code from mnt_oauth needs review

* [New] OAuth2 Doctypes and endpoints added

Integrations > OAuth Provider Settings
Integration Broker > OAuth Bearer Token
Integration Broker > OAuth Authorization Code
Integration Broker > OAuth Client
endpoints for authorize, approve, get_token, revoke_token and test_resource

* oauth2.py: renamed skipauth to skip_auth

* [Fix] Parse URL instead of storing it in settings

* [Fix] get skip_auth from OAuth Provider Settings

* Success URL format. Failure URL added. Confirmation dialog layout changed.

* Validate access token if passed during use of REST API

* OAuth Confirmation colours fixed

* Multiple Changes

Added links for OAuth under Integrations in Module list.
Updated permissions on OAuth doctypes.
Updated layout of OAuth Client doctype.

* [Docs] Integrations > How to setup OAuth

* [Docs] Integration > Using OAuth

* [Fix] get_token endpoint must to handle POST request

* [Fix] http verbs and responses for OAuth 2.0 Endpoints

* [Fix] accept oauth2 access_token from headers

* Removed unused imports from api.py

neilLasrado committed:
[fix] Quick entry from grid view (#2194)

rmehta committed:
[minor] dont show disabled roles in permission manager (#2180)

KanchanChauhan committed:
[WIP] Option to align field in Print Format Builder (#2109)

netchampfaris committed:
[ui-fix] ui-autocomplete z-index fix (#2191)

saurabh6790 committed:
[fix] page break fix in print format (#2193)

rmehta commented:
@bcornwellmott Can you add a property in DocType Report to show totals that can be saved? Or maybe I can assign someone else to do this.

I think without this, the feature will be incomplete

saurabh6790 committed:
[fix] typo fix

rohitwaghchaure committed:
Added setup_field_label_map method in utils (#2178)

saurabh6790 committed:
Merge pull request #2197 from saurabh6790/integration_fix

[fix] Scheduler fixes for Integration Services

rmehta committed:
[fix] user name must also be a valid email (#2196)

bcornwellmott committed:
Using Add Total Row checkbox in Report doctype

bcornwellmott committed:
Merge branch 'add_totals_row_to_report' of https://github.com/bcornwellmott/frappe into add_totals_row_to_report

Conflicts:
	frappe/desk/reportview.py
	frappe/public/js/frappe/views/reports/reportview.js

bcornwellmott committed:
Uses checkbox from Report add_total_row

netchampfaris committed:
[fix] Help link alignment

KanchanChauhan committed:
Disabled reports will not be searchable from Awesome Bar

nabinhait committed:
minor fix

nabinhait committed:
Change Logs for v7.1

nabinhait committed:
Merge pull request #2200 from KanchanChauhan/diabled-report-hide-awesome-bar

Disabled reports will not be searchable from Awesome Bar

nabinhait committed:
Merge pull request #2199 from netchampfaris/ui-fixes

[ui-fix] Help link alignment #2046

nabinhait committed:
Merge pull request #2201 from nabinhait/develop

Change Logs for v7.1

nabinhait committed:
bumped to version 7.0.47

nabinhait committed:
Merge branch 'hotfix'

nabinhait committed:
Fixed version conflict

rmehta committed:
[cleanup] dashboard filters

rmehta committed:
[minor] fix style for no data

rmehta committed:
[fix] default for checks in db_insert and show top 20 filter values

rmehta committed:
[fix] default value for check

rmehta committed:
[fix] in_standard_filter should be in docfield, not doctype

nabinhait committed:
[translations]

nabinhait committed:
Merge branch 'develop'

nabinhait committed:
bumped to version 7.2.0

nabinhait committed:
Update __init__.py

nabinhait committed:
Merge branch 'develop'

nabinhait committed:
bumped to version 7.1.0

bcornwellmott commented:
I think I screwed this branch up. Will close this PR and start a new one
